ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -1026,4 +1037,32 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
   var isSystemKeyboardTopBarEnabled: Bool {
     return true
   }
+  
+  var vibrationSupportLevel: VibrationSupport {
+    let device = Device()
+
+    if device.isPod {
+      return .none
+    } else if device.isPad {
+      // May not be entirely true, but I can't find any documentation on which
+      // ones DO have it.  Closest thing is https://discussions.apple.com/thread/7415858
+      // that suggests none before Jan 2016 had it, at least.
+      return .none
+    } else if device.isPhone {
+      let basicVibrationModels: [Device] = [.iPhone4, .iPhone4s, .iPhone5, Device.iPhone5s,
```

Comment:
Note to self:  this list needs to include the iPhone SE - it doesn't have the Taptic engine:  https://9to5mac.com/2016/03/22/comparison-iphone-se-6-6s-plus-specs/